evalml/automl/engine/engine_base.py

```diff
@@ -194,16 +210,14 @@ def train_and_score_pipeline(
         try:
             logger.debug(f"\t\t\tFold {i}: starting training")
             cv_pipeline = train_pipeline(
-                pipeline,
-                X_train,
-                y_train,
-                automl_config.optimize_thresholds,
-                automl_config.objective,
+                pipeline, X_train, y_train, automl_config, schema=False
             )
             logger.debug(f"\t\t\tFold {i}: finished training")
             if (
                 automl_config.optimize_thresholds
-                and pipeline.can_tune_threshold_with_objective(automl_config.objective)
+                and pipeline.can_tune_threshold_with_objective(
+                    threshold_tuning_objective
```

Comment:
@dsherry so rather than printing which fold, we just print that we found an optimal threshold whenever we call `train_pipeline`? This seems like unnecessary log statements if a user just called `train_pipelines` or something else outside of `AutoMLSearch.search()`.

Reply:
Yep, that's a good point.

My reasoning: if the choice is between duplicated code and extra debug-level log statements, personally I'd choose the latter, because duplicated code could lead to bugs later.

Reply:
@dsherry I think I'll leave it out for now since `train_pipeline` doesn't include a `logger` parameter, and I think this would be an easy change to address in the future if we wanted to refactor some code.